src/core/catalog.js

```diff
@@ -982,7 +982,9 @@ class Catalog {
       if (javaScript === null) {
         javaScript = new Map();
       }
-      javaScript.set(name, stringToPDFString(js));
+      js = stringToPDFString(js);
+      js = js.replace(/\u0000+/, "");
+      javaScript.set(name, js);
```

Comment:
Similar to the other comment:
```suggestion
      js = stringToPDFString(js).replace(/\u0000/g, "");
      javaScript.set(name, js);
```